src/agentscope/service/web_search/arxiv.py

```diff
@@ -0,0 +1,249 @@
+        "updated": _parse_timestamp(feed.updated_parsed),
+        "opensearch_total_results": int(feed.opensearch_totalresults),
+        "opensearch_start_index": int(feed.opensearch_startindex),
+        "opensearch_itemsperpage": int(feed.opensearch_itemsperpage),
+    }
+
+    # Entries
+    entries = []
+    for entry in result.entries:
+        title = "0"
+        if hasattr(entry, "title"):
+            title = entry.title
+        else:
+            logger.warning(
+                "Result %s is missing title attribute; defaulting to '0'",
+                entry.id,
+            )
+
+        tags = [tag.get("term") for tag in entry.tags]
+        if len(tags) == 0:
+            tags = None
+
+        entry_dict = _Result(
+            # Basic properties
+            entry_id=entry.id,
+            title=title,
+            updated=_parse_timestamp(entry.updated_parsed),
+            published=_parse_timestamp(entry.published_parsed),
+            summary=entry.summary,
+            authors=[author.name for author in entry.authors],
+            # Optional properties
+            pdf_url=_parse_pdf_url(entry.links),
+            comment=entry.get("arxiv_comment"),
+            primary_category=entry.arxiv_primary_category.get("term"),
+            tags=tags,
+            journal_ref=entry.get("arxiv_journal_ref"),
+            doi=entry.get("arxiv_doi"),
+        )
+
+        entries.append(entry_dict)
+
+    cleaned_dict["entries"] = entries
+
+    return cleaned_dict
+
+
+def arxiv_search(
+    search_query: str,
+    id_list: List[str] = None,
+    start: int = 0,
+    max_results: Optional[int] = None,
+) -> ServiceResponse:
+    """Search arXiv paper by a given query string.
+
+    Args:
+        search_query (`str`):
+            The query string, supporting prefixes "all:", "ti:", "au:",
+            "abs:", "co:", "jr:", "cat:", and "rn:", boolean operators "AND",
+            "OR" and "ANDNOT". For example, searching for papers with
+            title "Deep Learning" and author "LeCun" by a
+            search_query "ti:Deep Learning AND au:LeCun".
+        id_list (`List[str]`, defaults to `None`):
+            A list of arXiv IDs to search.
+        start (`int`, defaults to `0`):
+            The index of the first search result to return.
+        max_results (`Optional[int]`, defaults to `None`):
+            The maximum number of search results to return.
+
+    Returns:
+        `ServiceResponse`: A dictionary with two variables: `status` and
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is a list of search results or error information,
+        which depends on the `status` variable.
+    """
+
+    if feedparser is None:
+        raise ImportError(
+            "The `feedparser` module is not installed. Please install it by "
+            "running `pip install feedparser`.",
+        )
+
+    # construct url
+    search_query = search_query.replace(" ", "+")
```

Comment:
If replaced like this, `ti:Deep Learning AND au:LeCun` will become `ti:Deep+Learning+AND+au:LeCun`, but
`ti:"Deep Learning"+AND+au:LeCun` is expected.

Reply:
Done, and a new unit test for the function `_reformat_query` is added.